### PR TITLE
feat added add to cart

### PR DESCRIPTION
**Pull request by AmishMishra11**
Added add to cart functionality

- add to cart will redirect to the login page if a user is not logged in.
- add to cart will add the item in the cart page
- if the item is already in the cart it will increase the quantity of the item
- added final bill functionality

To test for yourself use the following email and password:
  email: "test@gmail.com"
  password: "test"

https://user-images.githubusercontent.com/76205249/159625854-5fa566d5-1a59-422e-bcfe-64e8f65b81a9.mp4

### CODE REVIEW

File: src/Components/Card/index.jsx [hunk 1]
```diff
@@ -2,23 +2,29 @@ import axios from "axios";
 import React from "react";
 import "./styles.css";
 import { useNavigate } from "react-router-dom";
```
Comment (SukanyaSen530) [inline src/Components/Card/index.jsx:4]: Suggestion - All these files can be imported into a index.js and imported together. Code look's cleaner and easy to read.

File: src/Components/Card/index.jsx [hunk 2]
```diff
@@ -2,23 +2,29 @@ import axios from "axios";
...
+
 import { useAuth } from "../../Contexts/AuthContext";
 
 import { useCart } from "../../Contexts/CartContext";
```
Comment (SukanyaSen530) [inline src/Components/Card/index.jsx:8]: Suggestion - All these Api calls files can be imported into a index.js and imported together. Code look's cleaner and easy to read.

**Review by SukanyaSen530 — APPROVED** [review]

**Review by Mohitkumar33 — APPROVED** [review]

**Comment by Mohitkumar33** [comment]
code is clean

File: src/Components/Card/index.jsx (hunk 2 above)
Comment (AmishMishra11) [reply src/Components/Card/index.jsx:8]: Thanks for the suggestion i will do that in the next pr

**Pull request merged**